apps/roam/src/utils/initializeObserversAndListeners.ts

```diff
@@ -189,27 +189,48 @@ export const initObservers = async ({
   const discourseNodeSearchTriggerListener = (e: Event) => {
     const evt = e as KeyboardEvent;
     const target = evt.target as HTMLElement;
+
     if (document.querySelector(".discourse-node-search-menu")) return;
 
-    if (evt.key === "@" || (evt.key === "2" && evt.shiftKey)) {
-      if (
-        target.tagName === "TEXTAREA" &&
-        target.classList.contains("rm-block-input")
-      ) {
-        const textarea = target as HTMLTextAreaElement;
-        const location = window.roamAlphaAPI.ui.getFocusedBlock();
-        if (!location) return;
-
-        const cursorPos = textarea.selectionStart;
-        const isBeginningOrAfterSpace =
-          cursorPos === 0 ||
-          textarea.value.charAt(cursorPos - 1) === " " ||
-          textarea.value.charAt(cursorPos - 1) === "\n";
-        if (isBeginningOrAfterSpace) {
+    if (
+      target.tagName === "TEXTAREA" &&
+      target.classList.contains("rm-block-input")
+    ) {
+      const textarea = target as HTMLTextAreaElement;
+      const location = window.roamAlphaAPI.ui.getFocusedBlock();
```

Comment:
Ensure this is only called when `customTrigger` the `evt.key`.  Right now this is being called on every keystroke.

Reply:
i think it needs to check every keystroke because we're trying to match a pattern (the key could be multiple chars, similar to smartblock) instead of listening to just 1 key.

Reply:
ahh i see what you mean. yes good point. I moved getFocusedBlock() check to the last step right before rendering the search menu to optimize